TES import

buchanae opened the pull request:
Rough draft of pulling Funnel events from Kafka and writing them into Arachne.

Commit:
initial

Commit:
capture task created event

adamstruck commented on import.go:
```diff
@@ -0,0 +1,98 @@
+  "github.com/golang/protobuf/proto"
+  "github.com/bmeg/arachne/aql"
+  "github.com/ohsu-comp-bio/mortar/events"
+  "github.com/ohsu-comp-bio/tes"
```
you don't need to import this if your including the proto file and generated code in this repo

adamstruck commented:
Looks good! 👍 

Generally,  my concern is the proliferation of duplication of code that is starting to occur in these repos:

- `github.com/ohsu-comp-bio/funnel`
- `github.com/ohsu-comp-bio/tes`
- `github.com/ohsu-comp-bio/mortar`

I think we need to start thinking carefully how to split up things conceptually and make sure import things whenever possible.

Commit:
remove tes package

buchanae commented:
I removed the `tes` package, which was accidentally committed.

As soon as we resolve https://github.com/ohsu-comp-bio/funnel/issues/481 I'll remove some of the events package too.

adamstruck commented on Makefile:
```diff
@@ -0,0 +1,8 @@
+		-I ./ \
+    -I $$GOPATH/src/github.com/golang/protobuf/ptypes/struct/ \
+    -I $$GOPATH/src/github.com/golang/protobuf/ptypes/timestamp/ \
+    --go_out=Mtes/tes.proto=github.com/ohsu-comp-bio/tes:. \
```
this command should be updated now that you removed the tes.proto file

Commit:
remove tes protoc build from Makefile

Commit:
File, Tag, Image verts for Task

buchanae commented:
I've added new functionality to this PR, so please review again.

This creates additional vertex types for each task. Still need edges.

Commit:
task edges, cleanup, and dev docs

buchanae commented:
Ok, the latest commit is in decent shape:

- Added edges from task to tag, file, etc.
- Cleaned up client, vertex, edge code
- Cleaned up Kafka event reader
- Added dev docs to readme and helpers to Makefile

I didn't put a ton of thought into vertex/edge label names, GID formats, etc. I'm pretty sure I'm diverging from BMEG on that so far. Probably cleanup to be done there. Also, we should have a style guide for those things.

Commit:
Add cobra command, config, and flags

buchanae commented:
Added a cobra CLI. I'm done adding to this branch now.

Commit:
add common Go project dev utils, tidy, and lint

buchanae commented:
@adamstruck requested tests and the basics of a Go project, so I guess I'm not done yet.

Added tidy/lint in the last commit.

Commit:
add simple test for graph.NewEdge

Commit:
cmd/version: add version command

Commit:
build: tweak deps/install

Commit:
build: add release, docker

Commit:
docs: add badges to readme

Commit:
build: pin gometalinter v2 and install deps

Commit:
graph/tes: fix File label

Commit:
cmd/run: skip task input.Content inputs

Commit:
build: fix gometalinter install command

Commit:
cmd/run: remove debug logs

Commit:
import rate logger, and Tags instead of per-tag

Commit:
clean up utils. add ktl workflow/step.

Commit:
simplify code with Batch
